core/domain/email_manager.py

```diff
@@ -164,5 +178,71 @@ def send_post_signup_email(user_id):
         EMAIL_FOOTER.value)
 
     _send_email(
-        user_id, feconf.SYSTEM_COMMITTER_ID, email_models.INTENT_SIGNUP,
+        user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
         email_subject, email_body)
+
+
+def require_valid_intent(intent):
+    if intent not in feconf.VALID_MODERATOR_ACTIONS:
+        raise Exception('Unrecognized email intent: %s' % intent)
+
+
+def _get_email_config(intent):
```

Comment:
I don't see a difference between an 'action' and an 'intent' and you've already checked that the action is valid in controllers, so I think this can be omitted.